### ISSUE TITLE
CIC does not make any ingress config in vpx.

### ISSUE THREAD

**Issue by kuktaikn** [citrix/citrix-k8s-ingress-controller#65]
I am creating a single-tier topology with CIC with VPX.
I set it in the following order.
1. Creat Kubernetes Cluster
2. I deployed CIC using citrix-k8s-ingress-controller.yaml
[citrix-k8s-ingress-controller.yaml.txt](https://github.com/citrix/citrix-k8s-ingress-controller/files/3364687/citrix-k8s-ingress-controller.yaml.txt)
   -          - --feature-node-watch ->> true
   - So CIC created a POD Network routing table on the VPX.
3. I created nginx pod using following command.
   - kubectl run nginx --image=nginx --port=80 --replicas=4
4. I created nginx service using following command.
   - kubectl expose deployment nginx --type=NodePort --name=nginxservice
5. I created ingress following yaml
[ingresstest.yaml.txt](https://github.com/citrix/citrix-k8s-ingress-controller/files/3364689/ingresstest.yaml.txt)

However, CIC did not generate anything on the VPX.
There are nothing in LB server, service, vserver, CS Action, Policy, Vserver.
So I checked the ns.log on VPX.
In the ns.log, the CIC periodically logs in, logs out,
And sometimes created and deleted the DUMMY vserver.
So I delete ClusterRole, ClusterRoleBinding, ServiceAccount, CIC pod.
And redeploy CIC, but But still CIC did not create anything on the VPX.
Did I misconfigure something?
Or CIC have a problem with VPX?

kubernetes version : v1.15.0
VPX version : NS12.1: Build 52.15.nc

**Comment by kuktaikn** [comment]
I checked pod log and finded some errors
But I do not know if it is related to this issue.
![image](https://user-images.githubusercontent.com/52600640/60757397-faf39b00-a044-11e9-87d4-79a90590ce5b.png)

**Comment by apoorva-05** [comment]
Hi @kuktaikn ,

Looks like the citrix-k8s-ingress-controller.yaml used has ingress class specified and the same isn't mentioned in the ingresstest.yaml
Please add the following annotation in the ingresstest.yaml for citrix-ingress-controller to configure the ingress on the VPX.
**kubernetes.io/ingress.class: "citrix"**

You can refer the ingress class document to know more about it.
https://github.com/citrix/citrix-k8s-ingress-controller/blob/master/docs/configure/ingress-classes.md 

Also the error message you are seeing is due to a special character [¡°] present in place of double quotes in the ingresstest.yaml just before persistencetype 
**ingress.citrix.com/lbvserver: '{"citrix-svc":{"lbmethod":"LEASTCONNECTION", ¡°persistenceType":"SOURCEIP"}}'**
Replacing the above line with the following should solve the issue.
****ingress.citrix.com/lbvserver: '{"citrix-svc":{"lbmethod":"LEASTCONNECTION", "persistenceType":"SOURCEIP"}}'** 

Please let us know if this resolved the issue.

**Comment by kuktaikn** [comment]
Dear apoorva-05
Thank you for your comment.
I was able to solve the issue with your help.
There are two miss config.
1. kubernetes.io/ingress.class: "citrix" 
2. ingress.citrix.com/lbvserver: '{"citrix-svc":{"lbmethod":"LEASTCONNECTION", ¡°persistenceType":"SOURCEIP"}}'

I don't know why special character [¡°] is in ingress.yaml
I still can not see the special character in my notepad and it looks normal.
So I erased that line.
And add kubernetes.io/ingress.class: "citrix" .
Then this issue resolved.
Thank you.
